### PR TITLE
Extend the documentation of Custom Disk Volumes

### PR DESCRIPTION
**Pull request by dcermak**
Changes proposed in this pull request:
* Extend the documentation about custom disk volumes especially with respect to the `<systemdisk>` element

### CODE REVIEW

File: doc/source/building/working_with_images/custom_volumes.rst [hunk 1]
```diff
@@ -5,60 +5,75 @@ Custom Disk Volumes
...
 KIWI supports defining custom volumes by using the logical volume manager
 (LVM) for the Linux kernel or by setting volumes at filesystem level when
-filesystem supports it, like Btrfs.
+filesystem supports it (e.g. btrfs).
 
-Volumes are defined inside the KIWI description file, :file:`config.xml`,
-under the `type` element by setting a `systemdisk` section. The
-`systemdisk` element can include `volume` child elements.
+Volumes are defined in the KIWI description file :file:`config.xml`,
+using the `systemdisk` element, which itself is a child of the `type`
```
Comment (tomschr) [inline doc/source/building/working_with_images/custom_volumes.rst]: I would suggest to shorten the paragraph a bit:

```suggestion
using `systemdisk`. This element is a child of the `type`.
```

File: doc/source/building/working_with_images/custom_volumes.rst [hunk 2]
```diff
@@ -5,60 +5,75 @@ Custom Disk Volumes
...
+element. Volumes themselves are added via (multiple) `volume` child
```
Comment (tomschr) [inline doc/source/building/working_with_images/custom_volumes.rst]:
```suggestion
Volumes themselves are added via (multiple) `volume` child
```

File: doc/source/building/working_with_images/custom_volumes.rst [hunk 3]
```diff
@@ -5,60 +5,75 @@ Custom Disk Volumes
...
+   </image>
 
-  * `mountpoint`: Optional attribute that specifies the mountpoint path.
+Additional non-root volumes are created for each present `volume`
+element. Volume details can be defined by setting the following `volume`
+attributes:
 
-  * `size`: Optional attribute to set the size of the volume. If no suffix
-    (`M` or `G`) is used the value is considered to be in megabytes.
+- `name`: Required attribute representing a volume's name. In addition this
```
Comment (tomschr) [inline doc/source/building/working_with_images/custom_volumes.rst]:
```suggestion
- `name`: Required attribute representing the volume's name. Additionally, this
```

File: doc/source/building/working_with_images/custom_volumes.rst [hunk 4]
```diff
@@ -5,60 +5,75 @@ Custom Disk Volumes
...
-    in megabytes.
+     .. code-block:: xml
 
-  * `label`: Optional attribute to set filesystem label name of the volume.
+        <volume name="@root" size="4G"/>
 
-  * `copy_on_write`: Optional attribute to apply the filesystem
-    copy-on-write attribute for this volume.
+- `freespace`: Optional attribute defining the additional free space added
+  to the volume. If no suffix (`M` or `G`) is used the value is considered
```
Comment (tomschr) [inline doc/source/building/working_with_images/custom_volumes.rst]:
```suggestion
  to the volume. If no suffix (`M` or `G`) is used, the value is considered
```

File: doc/source/building/working_with_images/custom_volumes.rst [hunk 5]
```diff
@@ -5,60 +5,75 @@ Custom Disk Volumes
...
 
-   <type>
-     <systemdisk name="vgroup-name">
-       <volume name="@root" freespace="5G"/>
-       <volume name="home" size="40G"/>
-       <volume name="tmp" size="1024"/>
-     </systemdisk>
-   </type>
+.. warning::
+   Note that size attributes for filesystem volumes, as for btrfs, are
```
Comment (tomschr) [inline doc/source/building/working_with_images/custom_volumes.rst]: "Note" inside a warning is superfluous.

```suggestion
   The size attributes for filesystem volumes, as for btrfs, are
```

File: doc/source/building/working_with_images/custom_volumes.rst [hunk 6]
```diff
@@ -5,60 +5,75 @@ Custom Disk Volumes
...
+   ignored and have no effect.
+
+
+The `systemdisk` element additionally supports the following optional
+attributes:
+
+- `name`: The volume group name, by default `kiwiVG` is used. This setting
+  is only relevant for LVM volumes.
+
+- `preferlvm`: Prefer LVM even if the used filesystem has its own volume
```
Comment (tomschr) [inline doc/source/building/working_with_images/custom_volumes.rst]: Is the value a boolean datatype? Which values are allowed?

File: doc/source/building/working_with_images/custom_volumes.rst [hunk 7]
```diff
@@ -5,60 +5,75 @@ Custom Disk Volumes
...
+         <volume name="usr/lib" size="1G" label="library"/>
+         <volume name="@root" freespace="500M"/>
+         <volume name="etc_volume" mountpoint="etc" copy_on_write="false"/>
+         <volume name="bin_volume" size="all" mountpoint="/usr/bin"/>
+       </systemdisk>
+     </type>
+   </image>
 
-  * `mountpoint`: Optional attribute that specifies the mountpoint path.
+Additional non-root volumes are created for each present `volume`
```
Comment (tomschr) [inline doc/source/building/working_with_images/custom_volumes.rst]: IHMO, the "present" word is not needed:

```suggestion
Additional non-root volumes are created for each `volume`
```

File: doc/source/building/working_with_images/custom_volumes.rst (hunk 6 above)
Comment (dcermak) [reply doc/source/building/working_with_images/custom_volumes.rst]: It's a boolean, I'll make this clearer.

**Pull request merged**